diff --git src/GitSHA1.cpp.in src/GitSHA1.cpp
similarity index 50%
rename from src/GitSHA1.cpp.in
rename to src/GitSHA1.cpp
index d073d1b68..0fd445fa8 100644
--- src/GitSHA1.cpp.in
+++ src/GitSHA1.cpp
@@ -20,48 +20,48 @@ OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 THE SOFTWARE.
 ***********************************************/
 
-#include "src/GitSHA1.h"
+#include "GitSHA1.h"
 
 const char* CMSat::get_version_sha1()
 {
-    static const char myversion_sha1[] = "@GIT_SHA1@";
+    static const char myversion_sha1[] = "f8b1da0eed202953912ff8cca10175eab61c0a1d";
     return myversion_sha1;
 }
 
 const char* CMSat::get_version_tag()
 {
-    static const char myversion_tag[] = "@PROJECT_VERSION@";
+    static const char myversion_tag[] = "5.8.0";
     return myversion_tag;
 }
 
 const char* CMSat::get_compilation_env()
 {
     static const char compilation_env[] =
-    "CMAKE_CXX_COMPILER = @CMAKE_CXX_COMPILER@ | "
-    "CMAKE_CXX_FLAGS = @CMAKE_CXX_FLAGS@ | "
-    "COMPILE_DEFINES = @COMPILE_DEFINES@ | "
-    "STATICCOMPILE = @STATICCOMPILE@ | "
-    "ONLY_SIMPLE = @ONLY_SIMPLE@ | "
-    "Boost_FOUND = @Boost_FOUND@ | "
-    "STATS = @STATS@ | "
-    "SQLITE3_FOUND = @SQLITE3_FOUND@ | "
-    "ZLIB_FOUND = @ZLIB_FOUND@ | "
-    "VALGRIND_FOUND = @VALGRIND_FOUND@ | "
-    "ENABLE_TESTING = @ENABLE_TESTING@ | "
-    "M4RI_FOUND = @M4RI_FOUND@ | "
-    "NOM4RI = @NOM4RI@ | "
-    "SLOW_DEBUG = @SLOW_DEBUG@ | "
-    "ENABLE_ASSERTIONS = @ENABLE_ASSERTIONS@ | "
-    "PYTHON_EXECUTABLE = @PYTHON_EXECUTABLE@ | "
-    "PYTHON_LIBRARY = @PYTHON_LIBRARY@ | "
-    "PYTHON_INCLUDE_DIRS = @PYTHON_INCLUDE_DIRS@ | "
-    "MY_TARGETS = @MY_TARGETS@ | "
-    "LARGEMEM = @LARGEMEM@ | "
-    "LIMITMEM = @LIMITMEM@ | "
-    "BREAKID_LIBRARIES = @BREAKID_LIBRARIES@ | "
-    "BREAKID-VER = @BREAKID_VERSION_MAJOR@.@BREAKID_VERSION_MINOR@ | "
-    "BOSPHORUS_LIBRARIES = @BOSPHORUS_LIBRARIES@ | "
-    "BOSPH-VER = @BOSPHORUS_VERSION_MAJOR@.@BOSPHORUS_VERSION_MINOR@ | "
+    "CMAKE_CXX_COMPILER = /usr/bin/c++ | "
+    "CMAKE_CXX_FLAGS =  -fvisibility=hidden -Wall -Wextra -Wunused -Wsign-compare -fno-omit-frame-pointer -Wtype-limits -Wuninitialized -Wno-deprecated -Wstrict-aliasing -Wpointer-arith -Wpointer-arith -Wformat-nonliteral -Winit-self -Wparentheses -Wunreachable-code -g -Wno-class-memaccess -mpopcnt -msse4.2 -ggdb3 -Wlogical-op -Wrestrict -Wnull-dereference -Wdouble-promotion -Wshadow -Wformat=2 -Wextra-semi -pedantic | "
+    "COMPILE_DEFINES =  -DEXTENDED_FEATURES -DUSE_GAUSS -DBOOST_TEST_DYN_LINK -DUSE_SQLITE3 -DUSE_ZLIB -DYALSAT_FPU | "
+    "STATICCOMPILE = OFF | "
+    "ONLY_SIMPLE = ON | "
+    "Boost_FOUND = FALSE | "
+    "STATS = OFF | "
+    "SQLITE3_FOUND = TRUE | "
+    "ZLIB_FOUND = TRUE | "
+    "VALGRIND_FOUND =  | "
+    "ENABLE_TESTING = OFF | "
+    "M4RI_FOUND = FALSE | "
+    "NOM4RI = OFF | "
+    "SLOW_DEBUG = OFF | "
+    "ENABLE_ASSERTIONS = ON | "
+    "PYTHON_EXECUTABLE = /usr/bin/python3 | "
+    "PYTHON_LIBRARY = /usr/lib/x86_64-linux-gnu/libpython3.8.so | "
+    "PYTHON_INCLUDE_DIRS = /usr/include/python3.8 | "
+    "MY_TARGETS =  | "
+    "LARGEMEM = OFF | "
+    "LIMITMEM = OFF | "
+    "BREAKID_LIBRARIES =  | "
+    "BREAKID-VER = . | "
+    "BOSPHORUS_LIBRARIES =  | "
+    "BOSPH-VER = . | "
     "compilation date time = " __DATE__ " " __TIME__
     ""
     ;
diff --git src/cryptominisat.h.in include/cryptominisat5/cryptominisat.h
similarity index 98%
rename from src/cryptominisat.h.in
rename to include/cryptominisat5/cryptominisat.h
index a37498416..fda1eee1f 100644
--- src/cryptominisat.h.in
+++ include/cryptominisat5/cryptominisat.h
@@ -23,9 +23,9 @@ THE SOFTWARE.
 #ifndef __CRYPTOMINISAT5_H__
 #define __CRYPTOMINISAT5_H__
 
-#define CRYPTOMINISAT_VERSION_MAJOR @PROJECT_VERSION_MAJOR@
-#define CRYPTOMINISAT_VERSION_MINOR @PROJECT_VERSION_MINOR@
-#define CRYPTOMINISAT_VERSION_PATCH @PROJECT_VERSION_PATCH@
+#define CRYPTOMINISAT_VERSION_MAJOR 5
+#define CRYPTOMINISAT_VERSION_MINOR 8
+#define CRYPTOMINISAT_VERSION_PATCH 0
 
 #include <atomic>
 #include <vector>
diff --git src/cryptominisat_c.h.in include/cryptominisat5/cryptominisat_c.h
similarity index 100%
rename from src/cryptominisat_c.h.in
rename to include/cryptominisat5/cryptominisat_c.h
diff --git src/solvertypesmini.h.in include/cryptominisat5/solvertypesmini.h
similarity index 100%
rename from src/solvertypesmini.h.in
rename to include/cryptominisat5/solvertypesmini.h
