diff --git src/GitSHA1.cpp.in src/GitSHA1.cpp
similarity index 50%
rename from src/GitSHA1.cpp.in
rename to src/GitSHA1.cpp
index d073d1b68..0fd445fa8 100644
--- src/GitSHA1.cpp.in
+++ src/GitSHA1.cpp
@@ -20,48 +20,48 @@ OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 THE SOFTWARE.
 ***********************************************/
 
-#include "src/GitSHA1.h"
+#include "GitSHA1.h"
 
 const char* CMSat::get_version_sha1()
 {
-    static const char myversion_sha1[] = "@GIT_SHA1@";
+    static const char myversion_sha1[] = "f8b1da0eed202953912ff8cca10175eab61c0a1d";
     return myversion_sha1;
 }
 
 const char* CMSat::get_version_tag()
 {
-    static const char myversion_tag[] = "@PROJECT_VERSION@";
+    static const char myversion_tag[] = "5.8.0";
     return myversion_tag;
 }
 
 const char* CMSat::get_compilation_env()
 {
     static const char compilation_env[] =
-    "CMAKE_CXX_COMPILER = @CMAKE_CXX_COMPILER@ | "
-    "CMAKE_CXX_FLAGS = @CMAKE_CXX_FLAGS@ | "
-    "COMPILE_DEFINES = @COMPILE_DEFINES@ | "
-    "STATICCOMPILE = @STATICCOMPILE@ | "
-    "ONLY_SIMPLE = @ONLY_SIMPLE@ | "
-    "Boost_FOUND = @Boost_FOUND@ | "
-    "STATS = @STATS@ | "
-    "SQLITE3_FOUND = @SQLITE3_FOUND@ | "
-    "ZLIB_FOUND = @ZLIB_FOUND@ | "
-    "VALGRIND_FOUND = @VALGRIND_FOUND@ | "
-    "ENABLE_TESTING = @ENABLE_TESTING@ | "
-    "M4RI_FOUND = @M4RI_FOUND@ | "
-    "NOM4RI = @NOM4RI@ | "
-    "SLOW_DEBUG = @SLOW_DEBUG@ | "
-    "ENABLE_ASSERTIONS = @ENABLE_ASSERTIONS@ | "
-    "PYTHON_EXECUTABLE = @PYTHON_EXECUTABLE@ | "
-    "PYTHON_LIBRARY = @PYTHON_LIBRARY@ | "
-    "PYTHON_INCLUDE_DIRS = @PYTHON_INCLUDE_DIRS@ | "
-    "MY_TARGETS = @MY_TARGETS@ | "
-    "LARGEMEM = @LARGEMEM@ | "
-    "LIMITMEM = @LIMITMEM@ | "
-    "BREAKID_LIBRARIES = @BREAKID_LIBRARIES@ | "
-    "BREAKID-VER = @BREAKID_VERSION_MAJOR@.@BREAKID_VERSION_MINOR@ | "
-    "BOSPHORUS_LIBRARIES = @BOSPHORUS_LIBRARIES@ | "
-    "BOSPH-VER = @BOSPHORUS_VERSION_MAJOR@.@BOSPHORUS_VERSION_MINOR@ | "
+    "CMAKE_CXX_COMPILER = /usr/bin/c++ | "
+    "CMAKE_CXX_FLAGS =  -fvisibility=hidden -Wall -Wextra -Wunused -Wsign-compare -fno-omit-frame-pointer -Wtype-limits -Wuninitialized -Wno-deprecated -Wstrict-aliasing -Wpointer-arith -Wpointer-arith -Wformat-nonliteral -Winit-self -Wparentheses -Wunreachable-code -g -Wno-class-memaccess -mpopcnt -msse4.2 -ggdb3 -Wlogical-op -Wrestrict -Wnull-dereference -Wdouble-promotion -Wshadow -Wformat=2 -Wextra-semi -pedantic | "
+    "COMPILE_DEFINES =  -DEXTENDED_FEATURES -DUSE_GAUSS -DBOOST_TEST_DYN_LINK -DUSE_SQLITE3 -DUSE_ZLIB -DYALSAT_FPU | "
+    "STATICCOMPILE = OFF | "
+    "ONLY_SIMPLE = ON | "
+    "Boost_FOUND = FALSE | "
+    "STATS = OFF | "
+    "SQLITE3_FOUND = TRUE | "
+    "ZLIB_FOUND = TRUE | "
+    "VALGRIND_FOUND =  | "
+    "ENABLE_TESTING = OFF | "
+    "M4RI_FOUND = FALSE | "
+    "NOM4RI = OFF | "
+    "SLOW_DEBUG = OFF | "
+    "ENABLE_ASSERTIONS = ON | "
+    "PYTHON_EXECUTABLE = /usr/bin/python3 | "
+    "PYTHON_LIBRARY = /usr/lib/x86_64-linux-gnu/libpython3.8.so | "
+    "PYTHON_INCLUDE_DIRS = /usr/include/python3.8 | "
+    "MY_TARGETS =  | "
+    "LARGEMEM = OFF | "
+    "LIMITMEM = OFF | "
+    "BREAKID_LIBRARIES =  | "
+    "BREAKID-VER = . | "
+    "BOSPHORUS_LIBRARIES =  | "
+    "BOSPH-VER = . | "
     "compilation date time = " __DATE__ " " __TIME__
     ""
     ;
diff --git src/cryptominisat.h include/cryptominisat5/cryptominisat.h
similarity index 100%
rename from src/cryptominisat.h
rename to include/cryptominisat5/cryptominisat.h
diff --git src/cryptominisat_c.h include/cryptominisat5/cryptominisat_c.h
similarity index 100%
rename from src/cryptominisat_c.h
rename to include/cryptominisat5/cryptominisat_c.h
diff --git src/solvertypesmini.h include/cryptominisat5/solvertypesmini.h
similarity index 100%
rename from src/solvertypesmini.h
rename to include/cryptominisat5/solvertypesmini.h
diff -bBdNrw -U5 --exclude=*.orig --exclude=.git --exclude=.svn --exclude=esesc.xcode --exclude=build --exclude=CVS --exclude=CVSmain --exclude=cscope.files --exclude=cscope.out --exclude=ChangeLog cryptominisat-5.11.11/src/satzilla_features.h cryptominisat-5.11.11-path/src/satzilla_features.h
--- src/satzilla_features.h	2023-06-25 07:10:12
+++ src/satzilla_features.h	2023-08-10 10:05:11
@@ -21,20 +21,19 @@
 ***********************************************/
 
 #ifndef SOLVE_FEATURES_H_
 #define SOLVE_FEATURES_H_
 
-#include <limits>
+#include <cmath>
 #include <cstdint>
+#include <limits>
 #include <string>
-#include <cmath>
 #include "constants.h"
 
 namespace CMSat {
 
-struct SatZillaFeatures
-{
+struct SatZillaFeatures {
     void print_stats() const;
 
     //Some parameter
     double eps = 0.00001;
 
@@ -47,36 +46,36 @@
     //Clause distribution
     double binary = 0;
     double horn = 0;
     double horn_mean = 0;
     double horn_std = 0;
-    double horn_min = numeric_limits<double>::max();
-    double horn_max = numeric_limits<double>::min();
+    double horn_min = std::numeric_limits<double>::max();
+    double horn_max = std::numeric_limits<double>::min();
     double horn_spread = 0;
 
     double vcg_var_mean = 0;
     double vcg_var_std = 0;
-    double vcg_var_min = numeric_limits<double>::max();
-    double vcg_var_max = numeric_limits<double>::min();
+    double vcg_var_min = std::numeric_limits<double>::max();
+    double vcg_var_max = std::numeric_limits<double>::min();
     double vcg_var_spread = 0;
 
     double vcg_cls_mean = 0;
     double vcg_cls_std = 0;
-    double vcg_cls_min = numeric_limits<double>::max();
-    double vcg_cls_max = numeric_limits<double>::min();
+    double vcg_cls_min = std::numeric_limits<double>::max();
+    double vcg_cls_max = std::numeric_limits<double>::min();
     double vcg_cls_spread = 0;
 
     double pnr_var_mean = 0;
     double pnr_var_std = 0;
-    double pnr_var_min = numeric_limits<double>::max();
-    double pnr_var_max = numeric_limits<double>::min();
+    double pnr_var_min = std::numeric_limits<double>::max();
+    double pnr_var_max = std::numeric_limits<double>::min();
     double pnr_var_spread = 0;
 
     double pnr_cls_mean = 0;
     double pnr_cls_std = 0;
-    double pnr_cls_min = numeric_limits<double>::max();
-    double pnr_cls_max = numeric_limits<double>::min();
+    double pnr_cls_min = std::numeric_limits<double>::max();
+    double pnr_cls_max = std::numeric_limits<double>::min();
     double pnr_cls_spread = 0;
 
     //Conflict clauses
     double avg_confl_size = 0.0;
     double confl_size_min = 0.0;
@@ -118,8 +117,8 @@
     //High-level satzilla_features
     uint64_t num_gates_found_last = 0;
     uint64_t num_xors_found_last = 0;
 };
 
-}
+} // namespace CMSat
 
 #endif //SOLVE_FEATURES_H_
