diff --git CMakeLists.txt CMakeLists.txt
index 42290c2..ed5cb6f 100644
--- CMakeLists.txt
+++ CMakeLists.txt
@@ -210,7 +210,9 @@ else()
   message(STATUS "Build STATIC library as: ${SLANG_LIB_NAME}")
 endif()

-include(external/CMakeLists.txt)
+#include(external/CMakeLists.txt)
+get_filename_component(FMT_ROOT_DIR "${FMT_CMAKE_PATH}" DIRECTORY)
+add_subdirectory("${FMT_ROOT_DIR}" fmt EXCLUDE_FROM_ALL)
 add_subdirectory(source)

 if(SLANG_INCLUDE_TESTS)
diff --git source/CMakeLists.txt source/CMakeLists.txt
index 6f3bb58..59ac3e7 100644
--- source/CMakeLists.txt
+++ source/CMakeLists.txt
@@ -143,7 +143,10 @@ target_include_directories(
   PUBLIC "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/../external/>")

 # Link libraries
-target_link_libraries(slang_slang PUBLIC fmt::fmt Boost::headers)
+target_link_libraries(slang_slang PUBLIC fmt::fmt)
+if(DEFINED ENV{EXT_BUILD_DEPS})
+  target_include_directories(slang_slang SYSTEM PUBLIC "$ENV{EXT_BUILD_DEPS}/include")
+endif()

 if(SLANG_USE_THREADS)
   find_package(Threads REQUIRED)
@@ -226,4 +229,7 @@ if(SLANG_INCLUDE_INSTALL)
     FILES ${CMAKE_CURRENT_BINARY_DIR}/slang/slang_export.h
     DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/slang/
     COMPONENT slang_Development)
+  install(FILES ../external/expected.hpp
+    DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}
+    COMPONENT slang_Development)
 endif()
