diff --git a/backends/cxxrtl/runtime/cxxrtl/capi/cxxrtl_capi.cc b/backends/cxxrtl/runtime/cxxrtl/capi/cxxrtl_capi.cc
index 34801c2d1..46e112989 100644
--- backends/cxxrtl/runtime/cxxrtl/capi/cxxrtl_capi.cc
+++ backends/cxxrtl/runtime/cxxrtl/capi/cxxrtl_capi.cc
@@ -18,8 +18,8 @@
 
 // This file is a part of the CXXRTL C API. It should be used together with `cxxrtl/capi/cxxrtl_capi.h`.
 
-#include <cxxrtl/capi/cxxrtl_capi.h>
-#include <cxxrtl/cxxrtl.h>
+#include "backends/cxxrtl/runtime/cxxrtl/capi/cxxrtl_capi.h"
+#include "backends/cxxrtl/runtime/cxxrtl/cxxrtl.h"
 
 struct _cxxrtl_handle {
 	std::unique_ptr<cxxrtl::module> module;
diff --git a/backends/cxxrtl/runtime/cxxrtl/capi/cxxrtl_capi_vcd.cc b/backends/cxxrtl/runtime/cxxrtl/capi/cxxrtl_capi_vcd.cc
index 0949a9363..37df39441 100644
--- backends/cxxrtl/runtime/cxxrtl/capi/cxxrtl_capi_vcd.cc
+++ backends/cxxrtl/runtime/cxxrtl/capi/cxxrtl_capi_vcd.cc
@@ -18,8 +18,8 @@
 
 // This file is a part of the CXXRTL C API. It should be used together with `cxxrtl/capi/cxxrtl_capi_vcd.h`.
 
-#include <cxxrtl/capi/cxxrtl_capi_vcd.h>
-#include <cxxrtl/cxxrtl_vcd.h>
+#include "backends/cxxrtl/runtime/cxxrtl/capi/cxxrtl_capi_vcd.h"
+#include "backends/cxxrtl/runtime/cxxrtl/cxxrtl_vcd.h"
 
 extern const cxxrtl::debug_items &cxxrtl_debug_items_from_handle(cxxrtl_handle handle);
 
diff --git a/backends/cxxrtl/runtime/cxxrtl/capi/cxxrtl_capi_vcd.h b/backends/cxxrtl/runtime/cxxrtl/capi/cxxrtl_capi_vcd.h
index 844f3cb8c..9963b8eda 100644
--- backends/cxxrtl/runtime/cxxrtl/capi/cxxrtl_capi_vcd.h
+++ backends/cxxrtl/runtime/cxxrtl/capi/cxxrtl_capi_vcd.h
@@ -27,7 +27,7 @@
 #include <stddef.h>
 #include <stdint.h>
 
-#include <cxxrtl/capi/cxxrtl_capi.h>
+#include "backends/cxxrtl/runtime/cxxrtl/capi/cxxrtl_capi.h"
 
 #ifdef __cplusplus
 extern "C" {
diff --git a/backends/cxxrtl/runtime/cxxrtl/cxxrtl.h b/backends/cxxrtl/runtime/cxxrtl/cxxrtl.h
index 511648cca..93f00a465 100644
--- backends/cxxrtl/runtime/cxxrtl/cxxrtl.h
+++ backends/cxxrtl/runtime/cxxrtl/cxxrtl.h
@@ -42,7 +42,7 @@
 #include <iostream>
 
 // `cxxrtl::debug_item` has to inherit from `cxxrtl_object` to satisfy strict aliasing requirements.
-#include <cxxrtl/capi/cxxrtl_capi.h>
+#include "backends/cxxrtl/runtime/cxxrtl/capi/cxxrtl_capi.h"
 
 #ifndef __has_attribute
 #	define __has_attribute(x) 0
diff --git a/backends/cxxrtl/runtime/cxxrtl/cxxrtl_replay.h b/backends/cxxrtl/runtime/cxxrtl/cxxrtl_replay.h
index 9aa3e105d..d4d41e437 100644
--- backends/cxxrtl/runtime/cxxrtl/cxxrtl_replay.h
+++ backends/cxxrtl/runtime/cxxrtl/cxxrtl_replay.h
@@ -32,8 +32,8 @@
 #include <atomic>
 #include <unordered_map>
 
-#include <cxxrtl/cxxrtl.h>
-#include <cxxrtl/cxxrtl_time.h>
+#include "backends/cxxrtl/runtime/cxxrtl/cxxrtl.h"
+#include "backends/cxxrtl/runtime/cxxrtl/cxxrtl_time.h"
 
 // Theory of operation
 // ===================
diff --git a/backends/cxxrtl/runtime/cxxrtl/cxxrtl_time.h b/backends/cxxrtl/runtime/cxxrtl/cxxrtl_time.h
index f37c2b656..b94e6a318 100644
--- backends/cxxrtl/runtime/cxxrtl/cxxrtl_time.h
+++ backends/cxxrtl/runtime/cxxrtl/cxxrtl_time.h
@@ -22,7 +22,7 @@
 #include <cinttypes>
 #include <string>
 
-#include <cxxrtl/cxxrtl.h>
+#include "backends/cxxrtl/runtime/cxxrtl/cxxrtl.h"
 
 namespace cxxrtl {
 
diff --git a/backends/cxxrtl/runtime/cxxrtl/cxxrtl_vcd.h b/backends/cxxrtl/runtime/cxxrtl/cxxrtl_vcd.h
index e8be70028..d0a663e34 100644
--- backends/cxxrtl/runtime/cxxrtl/cxxrtl_vcd.h
+++ backends/cxxrtl/runtime/cxxrtl/cxxrtl_vcd.h
@@ -19,7 +19,7 @@
 #ifndef CXXRTL_VCD_H
 #define CXXRTL_VCD_H
 
-#include <cxxrtl/cxxrtl.h>
+#include "backends/cxxrtl/runtime/cxxrtl/cxxrtl.h"
 
 namespace cxxrtl {
 
diff --git a/passes/equiv/equiv_make.cc b/passes/equiv/equiv_make.cc
index 602ad776d..3a5b2e61b 100644
--- passes/equiv/equiv_make.cc
+++ passes/equiv/equiv_make.cc
@@ -149,6 +149,18 @@ struct EquivMakeWorker
 
 		// list of cells without added $equiv cells
 		auto cells_list = equiv_mod->cells().to_vector();
+		std::set<const RTLIL::Wire *> signed_wire;
+
+		for (auto cell : cells_list) {
+			if (cell->type.in(ID($pos), ID($sshr), ID($add), ID($mul), ID($and), ID($or), ID($xor), ID($sub))) {
+				bool is_signed = cell->getParam(ID::A_SIGNED).as_bool(); // || cell->type == ID($sub);
+				if (is_signed) {
+					const RTLIL::Wire *wire = cell->getPort(ID::Y).chunks().at(0).wire;
+					if (wire->is_signed)
+						signed_wire.insert(wire);
+				}
+			}
+		}
 
 		for (auto id : wire_names)
 		{
@@ -225,7 +237,7 @@ struct EquivMakeWorker
 				gate_wire = dec_wire;
 			}
 
-			if (gold_wire == nullptr || gate_wire == nullptr || gold_wire->width != gate_wire->width) {
+			if (gold_wire == nullptr || gate_wire == nullptr) {
 				if (gold_wire && gold_wire->port_id)
 					log_error("Can't match gold port `%s' to a gate port.\n", gold_wire);
 				if (gate_wire && gate_wire->port_id)
@@ -237,6 +249,42 @@ struct EquivMakeWorker
 					gold_wire, log_signal(assign_map(gold_wire)),
 					gate_wire, log_signal(assign_map(gate_wire)), id.unescape());
 
+ 			if (gold_wire->width > gate_wire->width) {
+ 				char name[gate_wire->name.str().size()+32];
+
+ 				auto w2 = RTLIL::SigSpec(gate_wire);
+ 				if (signed_wire.count(gate_wire) || gate_wire->is_signed || gold_wire->is_signed) {
+ 					sprintf(name, "%s_sext", gate_wire->name.c_str());
+ 					w2.extend_u0(gold_wire->width, true);  // sign extend
+ 				}else{
+ 					sprintf(name, "%s_zext", gate_wire->name.c_str());
+ 					w2.extend_u0(gold_wire->width, false);  // zero extend
+ 				}
+
+ 				RTLIL::IdString id_name(name);
+ 				RTLIL::Wire *wire =  equiv_mod->addWire(id_name, gold_wire->width);
+ 				equiv_mod->connect(wire, w2);
+
+ 				gate_wire = wire;
+ 			}else if (gold_wire->width < gate_wire->width) {
+ 				char name[gate_wire->name.str().size()+32];
+
+ 				auto w2 = RTLIL::SigSpec(gold_wire);
+ 				if (signed_wire.count(gold_wire)) {
+ 					sprintf(name, "%s_sext", gold_wire->name.c_str());
+ 					w2.extend_u0(gate_wire->width, true);  // sign extend
+ 				}else{
+ 					sprintf(name, "%s_zext", gold_wire->name.c_str());
+ 					w2.extend_u0(gate_wire->width, false);  // zero extend
+ 				}
+
+ 				RTLIL::IdString id_name(name);
+ 				RTLIL::Wire *wire =  equiv_mod->addWire(id_name, gate_wire->width);
+ 				equiv_mod->connect(wire, w2);
+
+ 				gold_wire = wire;
+ 			}
+
 			if (gold_wire->port_output || gate_wire->port_output)
 			{
 				gold_wire->port_input = false;
diff --git a/passes/sat/miter.cc b/passes/sat/miter.cc
index 5dd1b07b4..21ec0b005 100644
--- passes/sat/miter.cc
+++ passes/sat/miter.cc
@@ -85,15 +85,14 @@ void create_miter_equiv(struct Pass *that, std::vector<std::string> args, RTLIL:
 	RTLIL::Module *gold_module = design->module(gold_name);
 	RTLIL::Module *gate_module = design->module(gate_name);
 	pool<Wire*> gold_cross_ports;
-
+	std::map<const RTLIL::Wire *, int> gold2resize;
 	for (auto gold_wire : gold_module->wires()) {
 		if (gold_wire->port_id == 0)
 			continue;
 		RTLIL::Wire *gate_wire = gate_module->wire(gold_wire->name);
 		if (gate_wire == nullptr)
 			goto match_gold_port_error;
-		if (gold_wire->width != gate_wire->width)
-			goto match_gold_port_error;
+
 		if (flag_cross && !gold_wire->port_input && gold_wire->port_output &&
 				gate_wire->port_input && !gate_wire->port_output) {
 			gold_cross_ports.insert(gold_wire);
@@ -103,6 +102,11 @@ void create_miter_equiv(struct Pass *that, std::vector<std::string> args, RTLIL:
 			goto match_gold_port_error;
 		if (gold_wire->port_output != gate_wire->port_output)
 			goto match_gold_port_error;
+		if (gold_wire->width != gate_wire->width) {
+			int delta = gold_wire->width;
+			delta -= gate_wire->width;
+			gold2resize[gold_wire] = delta;
+		}
 		continue;
 	match_gold_port_error:
 		log_cmd_error("No matching port in gate module was found for %s!\n", gold_wire->name);
@@ -114,8 +118,6 @@ void create_miter_equiv(struct Pass *that, std::vector<std::string> args, RTLIL:
 		RTLIL::Wire *gold_wire = gold_module->wire(gate_wire->name);
 		if (gold_wire == nullptr)
 			goto match_gate_port_error;
-		if (gate_wire->width != gold_wire->width)
-			goto match_gate_port_error;
 		if (flag_cross && !gold_wire->port_input && gold_wire->port_output &&
 				gate_wire->port_input && !gate_wire->port_output)
 			continue;
@@ -165,18 +167,58 @@ void create_miter_equiv(struct Pass *that, std::vector<std::string> args, RTLIL:
 			gold_cell->setPort(gold_wire->name, w);
 			gate_cell->setPort(gold_wire->name, w);
 		}
+		RTLIL::Wire *w_gold = nullptr;
 
 		if (gold_wire->port_output)
 		{
-			RTLIL::Wire *w_gold = miter_module->addWire("\\gold_" + gold_wire->name.unescape(), gold_wire->width);
-			w_gold->port_output = flag_make_outputs;
+			RTLIL::Wire *w_gold = nullptr;
+      		RTLIL::Wire *w_gate = nullptr;
+		#if 1
+      		auto it = gold2resize.find(gold_wire);
+		if (it != gold2resize.end()) {
+        	int delta = it->second;
+
+        	bool is_signed = gold_wire->is_signed;
+        	auto new_width = gold_wire->width + delta;
+        	if (delta<0)
+          		new_width = gold_wire->width - delta;
+
+       		if (delta<0) {
+        		// gold is smaller than gate
+				w_gold = miter_module->addWire("\\gold_tmp" + gold_wire->name.unescape(), gold_wire->width);
+           		gold_cell->setPort(gold_wire->name, w_gold);
+
+           		auto w2 = RTLIL::SigSpec(w_gold);
+           		auto nbits = -delta;
+           		for (int i=0;i<nbits;i++)
+           			w2.append(RTLIL::State::Sx);
+
+           		// to keep X at the upper bits w2.extend_u0(new_width, is_signed);
+
+           		RTLIL::Wire *new_wire = miter_module->addWire("\\gold_" + gold_wire->name.unescape(), new_width);
+
+           		miter_module->connect(new_wire, w2);
+           		w_gold = new_wire;
+				w_gold->port_output = flag_make_outputs;
+				w_gate = miter_module->addWire("\\gate_" + gold_wire->name.unescape(), gold_wire->width-delta);
+				w_gate->port_output = flag_make_outputs;
+				gate_cell->setPort(gold_wire->name, w_gate);
+			}
+		}
+		#endif
 
-			RTLIL::Wire *w_gate = miter_module->addWire("\\gate_" + gold_wire->name.unescape(), gold_wire->width);
-			w_gate->port_output = flag_make_outputs;
+      	if (w_gold==nullptr) {
+        	w_gold = miter_module->addWire("\\gold_" + gold_wire->name.unescape(), gold_wire->width);
+        	w_gold->port_output = flag_make_outputs;
 
 			gold_cell->setPort(gold_wire->name, w_gold);
-			gate_cell->setPort(gold_wire->name, w_gate);
+		}
+		if (w_gate==nullptr) {
+        	w_gate = miter_module->addWire("\\gate_" + gold_wire->name.unescape(), gold_wire->width);
+        	w_gate->port_output = flag_make_outputs;
 
+			gate_cell->setPort(gold_wire->name, w_gate);
+		}
 			RTLIL::SigSpec this_condition;
 
 			if (flag_ignore_gold_x)
diff --git passes/sat/async2sync.cc passes/sat/async2sync.cc
index be2355e00..808fdef3f 100644
--- passes/sat/async2sync.cc
+++ passes/sat/async2sync.cc
@@ -345,7 +345,8 @@ struct Async2syncPass : public Pass {
 						module->connect(ff.sig_q, new_d);
 					}
 
-					ff.sig_d = new_d;
+					// Store the visible latch output, including asynchronous reset/set.
+					ff.sig_d = ff.sig_q;
 					ff.sig_q = new_q;
 					ff.has_aload = false;
 					ff.has_arst = false;
